### ISSUE TITLE
Binding of a Claim to an application fails as Secret cannot be generated "illegal base64 data at input byte 3."

### ISSUE THREAD

**Issue by cmoulliard** [halkyonio/primaza-poc#177]
## Issue

Binding of a Claim to an application fails. Test here 
```
2022-11-30 14:40:58,478 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-10) HTTP Request to /applications/bind/5 failed, error id: 79673b21-8620-4d1c-9823
    at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:105)
    at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:359)
    at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
    at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
    at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
    at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
    at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
    at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
    at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
    at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
    at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
    at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:91)
    at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:564)
    at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
    at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
    at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
    at java.base/java.lang.Thread.run(Thread.java:833)
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://kubernetes.default.svc/api/v1/namespaces/app/secrets. Message: Secr
et in version "v1" cannot be handled as a Secret: illegal base64 data at input byte 3. Received status: Status(apiVersion=v1, code=400, details=null, kind=Status, message
=Secret in version "v1" cannot be handled as a Secret: illegal base64 data at input byte 3, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=nul
l, selfLink=null, additionalProperties={}), reason=BadRequest, status=Failure, additionalProperties={}).
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:684)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:664)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:615)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:558)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:521)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:308)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:644)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:83)
    at io.fabric8.kubernetes.client.dsl.base.CreateOnlyResourceOperation.create(CreateOnlyResourceOperation.java:61)
    at io.halkyon.services.KubernetesClientService.mountSecretInApplication(KubernetesClientService.java:66)
    at io.halkyon.services.KubernetesClientService_ClientProxy.mountSecretInApplication(Unknown Source)
    at io.halkyon.services.BindApplicationService.createSecretForApplication(BindApplicationService.java:48)
    at io.halkyon.services.BindApplicationService.bindApplication(BindApplicationService.java:30)
    at io.halkyon.services.BindApplicationService_ClientProxy.bindApplication(Unknown Source)
    at io.halkyon.resource.page.ApplicationResource.doBindApplication(ApplicationResource.java:59)
    at io.halkyon.resource.page.ApplicationResource_Subclass.doBindApplication$$superforward1(Unknown Source)
```

## How to reproduce

- Create a Claim

![Image](https://user-images.githubusercontent.com/463790/204826054-62f2746e-75be-42d6-9862-b0f4ae6dfa34.png)

- Bind it to an application
 
![Image](https://user-images.githubusercontent.com/463790/204825783-99e73e8e-3817-4332-a401-ec5dc27041ff.png)

**Comment by cmoulliard** [comment]
Should we convert the String to base64 ?
```
        secretData.put("url", Base64.getEncoder().encodeToString(url.getBytes()));
        secretData.put("username", Base64.getEncoder().encodeToString(credential.username.getBytes()));
        secretData.put("password", Base64.getEncoder().encodeToString(credential.password.getBytes()));
```
@aureamunoz

**Comment by cmoulliard** [comment]
Fix 269861f

**Comment by aureamunoz** [comment]
Ok
